### ISSUE TITLE
IllegalArgumentException with @Builder.Switch when using Eclipse JDT compiler

### ISSUE THREAD

**Issue by crumley** [immutables/immutables#238]
Immutables 2.1.7  / Eclipse Mars SR1

Compiling the following java code below will result in the except below (visible in the Eclipse view "Error Log" after saving the class or doing a clean build). 

The nasty part is this stops all further annotation processing for the rest of the project.

Repro class:
```java
@FactoryStyle
public class EclipseSwitchTest {
	public enum EnumForSwitch {
		YES, NO
	}

	@Builder.Factory
	public static ByteString factoryMethod(@Builder.Switch(defaultName = "YES") EnumForSwitch val) {
		return null;
	}
}
```

Stack Trace:
```
org.immutables.value.internal.$processor$.$Processor threw java.lang.IllegalArgumentException: Key 'public static EclipseSwitchTest.EnumForSwitch valueOf(java.lang.String) ' not present in map
	at org.immutables.value.internal.$guava$.base.$Preconditions.checkArgument($Preconditions.java:145)
	at org.immutables.value.internal.$guava$.base.$Functions$FunctionForMapNoDefault.apply($Functions.java:121)
	at org.immutables.value.internal.$guava$.base.$Functions$FunctionComposition.apply($Functions.java:216)
	at org.immutables.value.internal.$guava$.collect.$ByFunctionOrdering.compare($ByFunctionOrdering.java:46)
	at java.util.TimSort.countRunAndMakeAscending(TimSort.java:351)
	at java.util.TimSort.sort(TimSort.java:216)
	at java.util.Arrays.sort(Arrays.java:1435)
	at org.immutables.value.internal.$guava$.collect.$Ordering.immutableSortedCopy($Ordering.java:846)
	at org.immutables.value.internal.$generator$.$SourceOrdering.getEnclosedElements($SourceOrdering.java:84)
	at org.immutables.value.internal.$processor$.meta.$SwitcherModel.constructOptions($SwitcherModel.java:31)
	at org.immutables.value.internal.$processor$.meta.$SwitcherModel.<init>($SwitcherModel.java:25)
	at org.immutables.value.internal.$processor$.meta.$ValueAttribute.initFactoryParamsIfApplicable($ValueAttribute.java:1001)
	at org.immutables.value.internal.$processor$.meta.$ValueAttribute.initAndValidate($ValueAttribute.java:795)
	at org.immutables.value.internal.$processor$.meta.$FactoryMethodAttributesCollector.collect($FactoryMethodAttributesCollector.java:62)
	at org.immutables.value.internal.$processor$.meta.$ValueTypeComposer.compose($ValueTypeComposer.java:96)
	at org.immutables.value.internal.$processor$.meta.$Round.collectValues($Round.java:112)
	at org.immutables.value.internal.$processor$.$Processor.process($Processor.java:50)
	at org.immutables.value.internal.$generator$.$AbstractGenerator.process($AbstractGenerator.java:89)
	at org.eclipse.jdt.internal.compiler.apt.dispatch.RoundDispatcher.handleProcessor(RoundDispatcher.java:139)
	at org.eclipse.jdt.internal.compiler.apt.dispatch.RoundDispatcher.round(RoundDispatcher.java:121)
	at org.eclipse.jdt.internal.compiler.apt.dispatch.BaseAnnotationProcessorManager.processAnnotations(BaseAnnotationProcessorManager.java:159)
	at org.eclipse.jdt.internal.apt.pluggable.core.dispatch.IdeAnnotationProcessorManager.processAnnotations(IdeAnnotationProcessorManager.java:134)
	at org.eclipse.jdt.internal.compiler.Compiler.processAnnotations(Compiler.java:909)
	at org.eclipse.jdt.internal.compiler.Compiler.compile(Compiler.java:434)
	at org.eclipse.jdt.internal.core.builder.AbstractImageBuilder.compile(AbstractImageBuilder.java:367)
	at org.eclipse.jdt.internal.core.builder.BatchImageBuilder.compile(BatchImageBuilder.java:179)
	at org.eclipse.jdt.internal.core.builder.AbstractImageBuilder.compile(AbstractImageBuilder.java:304)
	at org.eclipse.jdt.internal.core.builder.BatchImageBuilder.build(BatchImageBuilder.java:61)
	at org.eclipse.jdt.internal.core.builder.JavaBuilder.buildAll(JavaBuilder.java:256)
	at org.eclipse.jdt.internal.core.builder.JavaBuilder.build(JavaBuilder.java:175)
	at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:734)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:205)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:245)
	at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:300)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:303)
	at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:359)
	at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:382)
	at org.eclipse.core.internal.events.AutoBuildJob.doBuild(AutoBuildJob.java:144)
	at org.eclipse.core.internal.events.AutoBuildJob.run(AutoBuildJob.java:235)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)
```

**Comment by elucash** [comment]
Thank you for reporting the issue! I don't know what actually incompatibility was, but the problematic call was replaced by a simpler one which worked (as I was able to reproduce the bug quite reliably). Please, try newly released 2.1.8 to see if problem gone.

**Comment by crumley** [comment]
No problem glad to help. 2.1.8 worked perfectly. Thanks!
